c/src/ml-api-service-agent-client.c

```diff
@@ -540,56 +540,61 @@ ml_service_model_get (const char *name, const unsigned int version,
   if (!result) {
     _ml_error_report ("Failed to invoke the method get_activated (%s).",
         err ? err->message : "Unknown error");
-    ret = ML_ERROR_IO_ERROR;
     g_clear_error (&err);
+    ret = ML_ERROR_IO_ERROR;
+    goto error;
   }
 
-  if (ML_ERROR_NONE != ret) {
-    g_free (description);
-    return ret;
+  if (ML_ERROR_NONE != ret || !description) {
+    goto error;
   }
 
   ret = ml_option_create (&_info);
   if (ML_ERROR_NONE != ret) {
-    g_free (description);
-    return ret;
+    goto error;
   }
 
   /* fill ml_info */
   parser = json_parser_new ();
   if (!parser) {
-    g_free (description);
-    _ml_error_report_return (ML_ERROR_OUT_OF_MEMORY,
-        "Failed to allocate memory for JsonParser. Out of memory?");
+    _ml_error_report
+        ("Failed to allocate memory for JsonParser. Out of memory?");
+    ret = ML_ERROR_OUT_OF_MEMORY;
+    goto error;
   }
 
   if (!json_parser_load_from_data (parser, description, -1, &err)) {
-    g_free (description);
-    g_error_free (err);
-    g_object_unref (parser);
-    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
-        "Failed to parse the json string. %s", err->message);
+    _ml_error_report ("Failed to parse the json string. %s", err->message);
+    g_clear_error (&err);
+    ret = ML_ERROR_INVALID_PARAMETER;
+    goto error;
   }
 
   root_node = json_parser_get_root (parser);
   if (!root_node) {
-    g_object_unref (parser);
-    g_free (description);
-    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
-        "Failed to get the root node of json string.");
+    _ml_error_report ("Failed to get the root node of json string.");
+    ret = ML_ERROR_INVALID_PARAMETER;
+    goto error;
   }
 
   j_object = json_node_get_object (root_node);
-  json_object_iter_init_ordered (&iter, j_object);
-  while (json_object_iter_next_ordered (&iter, &member_name, &member_node)) {
+  json_object_iter_init (&iter, j_object);
```

Comment:
I don't know why for sure, but in the Tizen-unified `..._ordered` should be used. Otherwise it malfunction.

Reply:
updated